src/main/scala/com/astrolabsoftware/grafink/CLParser.scala

```diff
@@ -42,7 +49,33 @@ trait CLParser {
           else failure("Option --config must be a valid file path")
         )
         .text("config accepts path to a configuration file")
+
+      opt[String]('s', "startdate")
+        .required()
+        .action((x, c) => c.copy(startDate = LocalDate.parse(x, dateFormat)))
+        .validate(x =>
+          if (validateDate(x)) success
+          else failure("Option --startdate must be a valid date in format (yyyy-MM-dd)")
+        )
+        .text("startdate accepts start day's date for which the job needs to be run (yyyy-MM-dd)")
+
+      opt[Int]('d', "duration")
+        .optional()
+        .action((x, c) => c.copy(duration = x))
+        .validate(x =>
+          if (x > 0 && x <= 30) success
```

Comment:
What is driving the max value for `duration`?

Reply:
Just the maximum amount of data that we want to process at a given time ?

Reply:
OK - so probably a week is more than enough (for LSST, we would have about a TB per night).

Reply:
ok , let me change this

Reply:
> Thanks, it looks good to me! I did not know ZIO yet, so it took me some time to explore it - nice!

I am exploring it properly myself :)